framework/db/ActiveRecord.php

```diff
@@ -455,18 +455,37 @@ protected function insertInternal($attributes = null)
             }
         }
         $db = static::getDb();
-        $command = $db->createCommand()->insert($this->tableName(), $values);
-        if (!$command->execute()) {
-            return false;
+        // depending on the underlying DBMS, either $returnParams will be not empty, the command will return results
+        // or lastInsertId has to be used to fetch primary key values
+        $returnParams = [];
+        $command = $db->createCommand()->insert($this->tableName(), $values, $this->primaryKey(), $returnParams);
+        // force preparing as a command for writing, not reading
```

Comment:
Because `queryOne` is called instead of `execute` which would prepare it as for reading and use slave connection.